.github/workflows/pull-request-workflow.yml

```diff
@@ -8,12 +8,12 @@ jobs:
 
     strategy:
       matrix:
-        node-version: [10.x]
+        node-version: [12.x]
 
     steps:
       - uses: actions/checkout@v2
         with:
-          ref: master
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
https://github.com/actions/checkout#checkout-pull-request-head-commit-instead-of-merge-commit